everett/manager.py

```diff
@@ -240,6 +241,100 @@ def __repr__(self):
         return '<ListOf(%s)>' % qualname(self.sub_parser)
 
 
+def _parse_rwx(modestr, default):
+    """parse out 'r', 'w' and 'x'.  use 'default' if none used or specified"""
+    mode = 0
+    if 'r' in modestr:
+        mode |= os.R_OK
+    if 'w' in modestr:
+        mode |= os.W_OK
+    if 'x' in modestr:
+        mode |= os.X_OK
+    if not mode:
+        mode = default
+    return mode
+
+
+def _mode2english(mode):
+    """turn a file mode into an english phrase"""
+    perms = []
+    if mode & os.R_OK: perms.append('read')
+    if mode & os.W_OK: perms.append('write')
+    if mode & os.X_OK: perms.append('execute')
+    return (','.join(perms[:-1]) + ' and ' if len(perms)>1 else '') + perms[-1]
+
+
+class PathValue(object):
+    """
+    A parser for paths
+
+    config('Pathname', parser=PathValue('w', isdir=True, exists=False))
+    """
+
+    def __init__(self, mode='', isdir=None, exists=True):
+        """
+        :arg mode: specifies if the file is any subset of readable ('r'), writable ('w') and executable ('x')
+        Ignores any other characters in 'mode'; defaults to 'r' if mode is unspecified, or 'rx' if isdir is True
+
+        :arg isdir: if True, check that it's a directory and not just any kind of file
+
+        :arg exists: if True, verify that the path exists.
+        """
+        self.isdir = isdir
+        self.exists = exists
+        self.mode = _parse_rwx(mode, os.R_OK|os.X_OK if isdir else os.R_OK)
+
+    def __call__(self, val):
+        """
+        :returns: (parser that returns) a normalized absolute path to the configured file
```

Comment:
I don't understand what this means. I suspect maybe removing the "`(parser that returns)`" part would make it clearer.

Reply:
:+1: